plugins/woocommerce-blocks/tests/e2e/tests/templates/single-product-template.block_theme.spec.ts

```diff
@@ -3,16 +3,69 @@
  */
 import { test, expect } from '@woocommerce/e2e-playwright-utils';
 
-// Sunglasses are defined as requering password in /bin/scripts/products.sh.
-const productWithPasswordPermalink = '/product/sunglasses/';
-
 test.describe( 'Single Product template', async () => {
 	test( 'shows password form in products protected with password', async ( {
 		page,
 	} ) => {
-		await page.goto( productWithPasswordPermalink );
+		// Sunglasses are defined as requiring password in /bin/scripts/products.sh.
+		await page.goto( '/product/sunglasses/' );
 		await expect(
 			page.getByText( 'This content is password protected.' ).first()
 		).toBeVisible();
 	} );
+
+	test( 'loads the Single Product template for a specific product', async ( {
+		admin,
+		editorUtils,
+		page,
+	} ) => {
+		const testData = {
+			productName: 'Belt',
+			permalink: '/product/belt',
+			templateName: 'Product: Belt',
+			templatePath: 'single-product-belt',
+			templateType: 'wp_template',
+		};
+		const userText = 'Hello World in the Belt template';
+
+		// Create the specific product template.
+		await admin.visitAdminPage(
+			'site-editor.php',
+			`path=/${ testData.templateType }`
+		);
+		await page.getByLabel( 'Add New Template' ).click();
+		await page
+			.getByRole( 'button', { name: 'Single item: Product' } )
+			.click();
+		await page
+			.getByPlaceholder( 'Search products' )
+			.fill( testData.productName );
+		await page
+			.getByRole( 'option', { name: testData.productName } )
+			.click();
+		await editorUtils.closeWelcomeGuideModal();
+		await page.getByLabel( 'Fallback content' ).click();
+
+		// Edit the template.
+		await editorUtils.editor.insertBlock( {
+			name: 'core/paragraph',
+			attributes: { content: userText },
+		} );
+		await editorUtils.saveTemplate();
+
+		// Verify edits are visible.
+		await page.goto( testData.permalink );
+		await expect( page.getByText( userText ).first() ).toBeVisible();
+
+		// Revert edition.
+		await admin.visitAdminPage(
+			'site-editor.php',
+			`path=/${ testData.templateType }/all`
+		);
+		await editorUtils.revertTemplateCreation( testData.templateName );
+		await page.goto( testData.permalink );
+
+		// Verify the edits are no longer visible.
+		await expect( page.getByText( userText ) ).toHaveCount( 0 );
```

Comment:
Is the purpose of this code to ensure the functionality of excluding the template, or is it just doing a test cleanup? If the latter, maybe it would be best to move this to an `afterAll` or `afterEach` hook. WDYT?

Reply:
It has a double purpose: to revert the edition for a test cleanup but also to verify that the "Revert edition" has worked correctly. Even though inside `editorUtils.revertTemplateCreation()` we check that a success notice is shown, I think it's a good idea to test that the template was, in fact, reverted.

I don't have a strong opinion about moving it into an `afterAll`/`afterEach`, but given that there is an `expect()`, I thought it made more sense to keep it inside the test? :thinking:

Reply:
That makes sense! I don't have a strong opinion on this either; I just pointed this out because if it was just a test cleanup and, for some reason, the test failed before the `editorUtils.revertTemplateCreation()` was called, then the templates will still exist after the test run, which could impact other tests that were not expecting these newly created templates.

Reply:
Good call! It happened to me a few times that a failing test messed up my test site and I had to restart it from scratch. So it's definitely something that would be worth investigating.

As we have some other test files with the same issue, I'll merge this PR as-is and investigate possible improvements in a separate PR, so we can tackle all test files at once. I opened an issue here: https://github.com/woocommerce/woocommerce/issues/44445.